CONTRIBUTING.md

```diff
@@ -17,11 +17,45 @@ git push -f
 ```
 
 It's okay to force push over your own PR branch. In fact, that's what we want so that the commits are clean.
-
+This also means you shouldn't use `git pull` and `git merge` when contributing.
 
 ## How to squash a PR
-See [this how to](https://twitter.com/stephenfluin/status/1009904095073718275).
+To squash multiple commits, use the following steps:
+
+### 1. Rebase multiple commits into one
+
+To know which state you should rebase from, use `git log --oneline` and copy the `[commit-hash]` where `[commit-hash]` is the hash of the commit just before the first one you want to rewrite from. In the example below, you should rebase from `287bc79` because it is the one you originally checked out from:
```

Comment:
I feel like these instructions may not quite be right. You should always be rebasing on `origin/main`.

Reply:
Oh, right. It will be always `origin/main` in any case.